### PR TITLE
Fixes for eslint issues encountered when initializing Portunus

### PR DESCRIPTION
**Pull request by Ian-MacLeod**
The for..of rule is one that we could consider disabling. This is a case where the airbnb rules are pretty opinionated - they prefer the use of array iteration methods like forEach.

### CODE REVIEW

**Comment by markrawls** [comment]
> The for..of rule is one that we could consider disabling.

I despise the existence of that rule and it baffles me that it ever made in.

`'no-restricted-syntax': ['off']` with extreme prejudice.

File: zygoat/components/frontend/resources/[...slug].js [hunk 1]
```diff
@@ -18,9 +18,8 @@ const fetcher = async (req, res) => {
     response = await axios[req.method.toLowerCase()](url, data, config);
   }
 
-  for (const [name, value] of Object.entries(response.headers)) {
-    res.setHeader(name, value);
-  }
+  const setHeader = ([name, value]) => res.setHeader(name, value)
+  Object.entries(response.headers).forEach(setHeader);
```
Comment (markrawls) [inline zygoat/components/frontend/resources/[...slug].js]: This is significantly less easy to read and AirBnB is wrong with their eslint rules. Nuke that rule from orbit and put it back.

File: zygoat/components/frontend/eslint/resources/.eslintrc.js [hunk 2]
```diff
@@ -45,6 +45,10 @@ module.exports = {
       }
     ],
 
+    // There are many situations where libraries expect you to modify the
+    //   properties of a parameter passed to a function.
+    'no-param-reassign': ['error', { props: false }],
+
```
Comment (markrawls) [inline zygoat/components/frontend/eslint/resources/.eslintrc.js]: A couple rules I also typically put in:

```
    'react/jsx-filename-extension': ['off'],
    'prettier/prettier': 'error',
    'react/jsx-props-no-spreading': ['off'],
    'func-names': ['off'],
    'react/forbid-prop-types': ['off'],
    'react/no-unescaped-entities': ['off'],

    // For components that appear more than once, this is impossible
    'jsx-a11y/click-events-have-key-events': ['off'],
    'jsx-a11y/no-static-element-interactions': ['off'],

    // I disagree with this in general
    'react/no-array-index-key': ['off'],
```

File: zygoat/components/frontend/eslint/resources/.eslintrc.js [hunk 3]
```diff
@@ -57,12 +61,14 @@ module.exports = {
...
-    "babel-module": {
-      root: ["./src"],
-      alias: {
-        "@wui": "@bequestinc/wui"
+  settings: {
+    "import/resolver": {
+      "babel-module": {
+        alias: {
+          "@wui": "@bequestinc/wui",
+          "@@": "/"
```
Comment (markrawls) [inline zygoat/components/frontend/eslint/resources/.eslintrc.js]: Should probably be `./` right?

File: zygoat/components/frontend/eslint/resources/.eslintrc.js (hunk 2 above)
Comment (Ian-MacLeod) [reply zygoat/components/frontend/eslint/resources/.eslintrc.js]: I agree or am okay with disabling pretty much everything here except that I think we should try to keep `func-names` on. Something that I find really annoying when debugging in the Willing app is that our stack traces have so many anonymous functions in them.

File: zygoat/components/frontend/resources/[...slug].js (hunk 1 above)
Comment (Ian-MacLeod) [reply zygoat/components/frontend/resources/[...slug].js]: Not sure that I agree on clarity in this specific case, but I do agree that there are cases where for...of is much more clear. Turning off the rule sounds good to me.

File: zygoat/components/frontend/eslint/resources/.eslintrc.js (hunk 2 above)
Comment (markrawls) [reply zygoat/components/frontend/eslint/resources/.eslintrc.js]: Fair enough. I don't feel incredibly strongly on anything except the array-index-key, so it's your call.

**Pull request merged**